lab-max/test/item-route-test.js

```diff
@@ -0,0 +1,426 @@
+'use strict';
+
+const PORT = process.env.PORT || 3000;
+process.env.MONGODB_URI = 'mongodb://localhost/storetest';
+
+const expect = require('chai').expect;
+const debug = require('debug')('store:item-test');
+const request = require('superagent');
+const Store = require('../model/store.js');
+const Item = require('../model/item.js');
+const server = require('../server.js');
+
+const url = `http://localhost:${PORT}`;
+
+const exampleItem = {
+  name: 'soccer shoes',
+  itemType: 'shoes',
+  price: 50,
+};
+
+const exampleStore = {
+  name: 'nike',
+  timestamp: new Date(),
+  year: 1964,
+  storeType: 'sporting goods',
+};
+
+describe('testing item routes', function(){
+
+  before((done) => {
+    debug('before module item-router-test');
+    if (! server.isRunning) {
+      server.listen(PORT, () => {
+        server.isRunning = true;
+        debug(`server up ::: ${PORT}`);
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  after((done) => {
+    debug('after module item-router-test');
```

Comment:
@meeksfred awesome use of stopping the server after running the associated tests - this is a great best practice